File: src/test/java/co/nlighten/jsontransform/functions/TransformerFunctionIsNullTest.java
package co.nlighten.jsontransform.functions;

import co.nlighten.jsontransform.BaseTest;
import com.google.gson.JsonNull;
import org.junit.jupiter.api.Test;

public class TransformerFunctionIsNullTest extends BaseTest {
    @Test
    void nullTest() {
        assertTransformation(null, "$$isnull:$", true);
        assertTransformation(JsonNull.INSTANCE, "$$isnull:$", true);
        assertTransformation(0, "$$isnull():$", false);
        assertTransformation("", "$$isnull:$", false);
        assertTransformation(false, "$$isnull:$", false);
    }
}

File: javascript/json-transform/src/__tests__/functions/TransformerFunctionIsNull.test.ts
import { describe, test } from "vitest";
import { assertTransformation } from "../BaseTransformationTest";

describe("TransformerFunctionIsNull", () => {
  test("nullTest", async () => {
    await assertTransformation(null, "$$isnull:$", true);
    await assertTransformation(undefined, "$$isnull:$", true);
    await assertTransformation(0, "$$isnull():$", false);
    await assertTransformation("", "$$isnull:$", false);
    await assertTransformation(false, "$$isnull:$", false);
  });
});
